tests/pytests/unit/states/test_user.py

```diff
@@ -264,3 +264,49 @@ def mock_exists(*args):
             "warndays": 7,
             "inactdays": 0,
         }
+
+
+def test_gecos_field_changes_in_user_present():
+    """
+    Test if the gecos fields change in salt.states.user.present
+    """
+    mock_info = MagicMock(
+        side_effect=[
+            {
+                "uid": 5000,
+                "gid": 5000,
+                "groups": ["foo"],
+                "home": "/home/foo",
+                "fullname": "Foo Bar",
+                "homephone": "667788",
+            },
+            {
+                "uid": 5000,
+                "gid": 5000,
+                "groups": ["foo"],
+                "home": "/home/foo",
+                "fullname": "Bar Bar",
+                "homephone": "44566",
+            },
+        ]
+    )
+    mock_changes = MagicMock(side_effect=[{"homephone": "667788"}, None])
+    shadow_info = MagicMock(
+        return_value={"min": 2, "max": 88888, "inact": 77, "warn": 14, "passwd": ""}
+    )
+    shadow_hash = MagicMock(return_value="abcd")
+    dunder_salt = {
+        "user.info": mock_info,
+        "shadow.info": shadow_info,
+        "shadow.default_hash": shadow_hash,
+        "file.group_to_gid": MagicMock(side_effect=["foo"]),
+        "file.gid_to_group": MagicMock(side_effect=[5000, 5000]),
+        "user.chhomephone": MagicMock(return_value=True),
+    }
```

Comment:
```suggestion
    dunder_salt = {
        "user.info": mock_info,
        "shadow.info": shadow_info,
        "shadow.default_hash": shadow_hash,
        "file.group_to_gid": MagicMock(side_effect=["foo"]),
        "file.gid_to_group": MagicMock(side_effect=[5000, 5000]),
        "user.chhomephone": MagicMock(return_value=True),
    }
```
I believe you can set the `__salt__` variables in the `configure_loader_modules` fixtures.
You can see an example in this file https://github.com/saltstack/salt/blob/master/tests/pytests/unit/modules/test_pip.py

Reply:
I'll move all the variables except `user.info `and `user.chhomephone` to the fixture, since the 2 I mentioned are specifically for the test I've created, so they won't be of much use outside the test. This is my thinking. I'll make a commit for you to check and you can comment on the same. Thank you

Reply:
@MKLeb I've made the change as suggested by you. Do let me know if anything is missing.